internal/pipeline/runtime/container.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"os/user"
+)
+
+// ContainerNetworkName is a type for network name used in container
+type containerNetworkName string
+
+const (
+	networkNameNone containerNetworkName = "none"
+	networkNameHost containerNetworkName = "host"
+)
+
+type containerFunctionPermission struct {
+	AllowNetwork  bool
+	AsCurrentUser bool
+}
+
+// ContainerRunner implements a KRMFn which run a containerized
+// KRM function
+type ContainerRunner struct {
+	// Image is the container image to run
+	Image string
+
+	Exec ExecRunner
```

Comment:
Not sure why we have this here ?

Reply:
1. This piece of code is modified from kyaml function runtime and the container runtime in kyaml is based on exec runtime.
2. Since we have plan to support exec function so I think it's better to have a separate runner for exec so I don't change this structure.

Reply:
My main motivation for that suggestion was decoupling the execFn and containerFn so that changes to one doesn't affect the another. Also note that the execFn is a very lightweight wrapper over golang's exec primitives, so this extra layering doesn't buy us much instead increases overhead for anyone understanding `containerFn`.

I know current code is inspired by the code in kyaml but we can diverge from that depending on our requirements.

> Since we have plan to support exec function so I think it's better to have a separate runner for exec so I don't change this structure.

Totally agree with having separate runner for execFn. I am suggesting even greater degree of separation between the two :)